src/query-builder/transformer/RawSqlResultsToEntityTransformer.ts

```diff
@@ -146,6 +147,36 @@ export class RawSqlResultsToEntityTransformer {
         return hasData;
     }
 
+    protected transformComputedColumns(rawResults: any[], alias: Alias, entity: ObjectLiteral): boolean {
+        let hasData = false;
+
+        this.expressionMap.computedSelects.forEach(computedSelect => {
+            if (computedSelect.aliasName && computedSelect.mapToProperty) {
+                const fields: any[] = computedSelect.mapToProperty.split("_");
+                let entityPart = entity;
+
+                if (fields[0] === alias.name) {
+                    fields.shift();
+                    fields.forEach((field, index) => {
+                        if (index < (fields.length - 1)) {
+                            if (field && !entityPart[field]) {
+                                entityPart[field] = {};
+                            }
+
+                            entityPart = entityPart[field];
+                        } else {
+                            const value = this.driver.prepareHydratedValue(rawResults[0][computedSelect.aliasName as string], computedSelect.columnType as string);
```

Comment:
Why accept many raw results if you only use the first?

Reply:
For the same reasons that I use `forEach`. The `transformColumns` method accepts rawResults but it only use the first:
```
const value = rawResults[0][DriverUtils.buildColumnAlias(this.driver, alias.name, column.databaseName)];
if (value === undefined || column.isVirtual)
return;
```

I am triying continue the developing styles of the project, but we can change this if you want.